src/ui/App.js

```diff
@@ -11,17 +11,53 @@ import { useState, useCallback } from "react";
 export default function App({ initialCards }) {
   const [cards, setCards] = useState(initialCards);
   const [selected, setSelected] = useState([]);
+  const [unselectedAcesCount, setUnselectedAcesCount] = useState(
+    numberOfUnselectedAces(cards)
+  );
+
+  const [newGame, setNewGame] = useState(false);
+
+  function numberOfUnselectedAces(cards) {
+    return cards.reduce((count, card) => {
+      return count + (card.rank === "A" && !card.selected ? 1 : 0);
+    }, 0);
+  }
 
   function toggleSelected(index) {
+    if (newGame) {
+      return;
+    }
+    // setUnselectedAcesCount(numberOfUnselectedAces(cards));
+
+    const isAce = cards[index].rank === "A";
+    console.log(unselectedAcesCount);
     if (!selected.includes(index)) {
-      setSelected(selected.concat([index]));
+      if (
+        selected.length < 3 || // Allow selecting up to 3 cards normally
+        (selected.length === 3 && unselectedAcesCount === 1 && !isAce) ||
+        unselectedAcesCount > 1
+      ) {
+        // Allow selecting the fourth card if the last unselected card is an ace
+        setSelected([...selected, index]);
+        if (isAce) {
+          setUnselectedAcesCount((prevCount) => prevCount - 1);
+        }
+      }
     } else {
-      setSelected(selected.filter((elt) => elt !== index));
+      setSelected(selected.filter((item) => item !== index));
+      if (isAce) {
+        setUnselectedAcesCount((prevCount) => prevCount + 1);
+      }
     }
   }
 
   // This function will be called when the Draw button is clicked
   const fetchNewCards = useCallback(async () => {
+    let s = selected.length;
+    if (newGame) {
```

Comment:
This is smart! But keep in mind if you play too many games you'll exhaust your deck. :)